### PR TITLE
[Fix] Fix publish failure

### PR DESCRIPTION
**Pull request by LazyBusyYang**
1. Fix .github/workflows/publish-to-pypi.yml:
    a. Do not build source_code.tar.gz.
    b. Repair wheel file with auditwheel and rename it from linux to manylinux2014, for twine upload
2. Docker build script, dockerfile and docker image for publish, based on [manylinux2014_x86_64](https://github.com/pypa/manylinux)
3. Remove `bdist_wheel` section from setup.cfg, because XRPrimer does not support Python 2 and it contains cpp extensions , see the [link](https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/?highlight=universal#wheels) for details.
4. Store cmake output lib dir to `self.cmake_output_lib`(temp/lib.linux-x86_64-cpython-38), instead of replacing `self.build_lib`(build/lib.linux-x86_64-cpython-38). build_lib is useful after calling cmake.
5. Copy xrprimer_cpp*.so from `self.cmake_output_lib` to `self.build_lib`, otherwise the library won't be injected into a wheel file, nor be moved to site-packages.
6. Adjust arugments for `setup()` function, accoding to [python doc](https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/).
7. Rebuild conan pacakges inside manylinux2014 docker and upload, deploy workflow(upload disabled) has passed, see [action](https://github.com/openxrlab/xrprimer/actions/runs/4782034268) for details.
8. Fix wrong calibration input variable reported by [issue-33](https://github.com/openxrlab/xrprimer/issues/33).
9. Update test data at [aliyun-oss](https://openxrlab-share-mainland.oss-cn-hangzhou.aliyuncs.com/xrprimer/xrprimer.tar.gz), fix wrong subfolder name which causes error.

### CODE REVIEW

File: dockerfiles/publish_manylinux/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    ln -s /usr/lib64/libpng15.so.15 /usr/local/lib/libpng.so && \
+    ln -s /usr/lib64/libtiff.so.5 /usr/local/lib/libtiff.so
+
+# Update in bashrc
+RUN echo "source /opt/miniconda/etc/profile.d/conda.sh" >> /root/.bashrc && \
+    echo "conda deactivate" >> /root/.bashrc
+
+# Set proxy
+ENV HTTPS_PROXY http://172.16.1.135:3128
+ENV HTTP_PROXY http://172.16.1.135:3128
```
Comment (yl-1993) [inline dockerfiles/publish_manylinux/Dockerfile:43]: Not necessary?

File: dockerfiles/publish_manylinux/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    pip cache purge
+
+# Prepare py-3.10 conda env
+RUN . /opt/miniconda/etc/profile.d/conda.sh && \
+    conda create -n py-3.10 python=3.10 -y && \
+    conda activate py-3.10 && \
+    pip install --upgrade pip setuptools wheel && \
+    pip install torch==1.11.0 torchvision==0.12.0  && \
+    pip install mmcv==1.7.0 && \
+    pip cache purge
```
Comment (yl-1993) [inline dockerfiles/publish_manylinux/Dockerfile:88]: What about `RUN` for-loop, and using `ARG` to define mmcv version?

Comment (LazyBusyYang) [reply dockerfiles/publish_manylinux/Dockerfile]: Loop added in https://github.com/openxrlab/xrprimer/pull/36/commits/df0a6792f11eb6a4e25a933d859623c882e25b91

File: dockerfiles/publish_manylinux/Dockerfile (hunk 1 above)
Comment (LazyBusyYang) [reply dockerfiles/publish_manylinux/Dockerfile]: Removed from dockerfile, but build history can be found at dockerhub:
https://hub.docker.com/layers/openxrlab/xrprimer_ci/manylinux2014_x86_64_torch1110_mmcv170/images/sha256-1a1b25dc3cefa40c832095dad4dc63bd9d4fd971d4467d5767e58352f4e439f1?context=repo

File: dockerfiles/publish_manylinux/Dockerfile (hunk 2 above)
Comment (LazyBusyYang) [reply dockerfiles/publish_manylinux/Dockerfile]: ARG added in [1a471be](https://github.com/openxrlab/xrprimer/pull/36/commits/1a471be99215559e80ce483ade7d140e15173b16)

File: dockerfiles/publish_manylinux/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    echo "conda deactivate" >> /root/.bashrc
+
+ARG MMCV_VER
+# Install py-3.6, py-3.7, py-3.8, py-3.9, py-3.10
+RUN . /opt/miniconda/etc/profile.d/conda.sh && \
+    for i in 6 7 8 9 10; do \
+        conda create -n "py-3.$i" python="3.$i" -y && \
+        conda activate "py-3.$i" && \
+        pip install --upgrade pip setuptools wheel && \
+        pip install torch==1.11.0 torchvision==0.12.0 && \
```
Comment (yl-1993) [inline dockerfiles/publish_manylinux/Dockerfile:43]: Use `ARG` for `torch` and `torchvision`

File: setup.py [hunk 4]
```diff
@@ -186,7 +186,7 @@ def build_extension(self, ext):
         # cpp/pybind/CMakeList.txt, thus we also set build_lib explicitly,
         # where self.build_lib is the directory for compiled extension modules
         build_temp = self.build_temp
-        self.build_lib = os.path.join(build_temp, 'lib')
+        self.cmake_output_lib = os.path.join(build_temp, 'lib')
         clean_cmake(folders=['_deps', '_exts', self.build_temp])
```
Comment (yl-1993) [inline setup.py:190]: Seems not '_ext' anymore

File: setup.py [hunk 5]
```diff
@@ -203,6 +203,18 @@ def build_extension(self, ext):
         # build project
         subprocess.check_call(['cmake', '-S.', '-B', build_temp] + cmake_args)
         subprocess.check_call(['cmake', '--build', build_temp] + build_args)
```
Comment (yl-1993) [inline setup.py:205]: Is it possible to merge L199-202 with L203-205?

File: dockerfiles/publish_manylinux/Dockerfile (hunk 3 above)
Comment (LazyBusyYang) [reply dockerfiles/publish_manylinux/Dockerfile]: Added in [c733dbf](https://github.com/openxrlab/xrprimer/pull/36/commits/c733dbfe4b048424977a150bbe0cd4abb9abe987)

File: setup.py (hunk 4 above)
Comment (LazyBusyYang) [reply setup.py:190]: Changed to `build_deps`

File: setup.py (hunk 5 above)
Comment (LazyBusyYang) [reply setup.py:207]: @GACLove

File: setup.py (hunk 4 above)
Comment (yl-1993) [reply setup.py:190]: Not commited yet? `clean_cmake(folders=['build_deps', self.build_temp])` works fine.

Comment (LazyBusyYang) [reply setup.py]: Updated in [18a8979](https://github.com/openxrlab/xrprimer/pull/36/commits/18a8979b073e750778ddfc51e393e9ff5c479705). Sry, only those `_ext` in cmake command are changed yesterday.

**Review by yl-1993 — APPROVED** [review]

**Pull request merged**